File: UNIVERSITY/Weathers_List/app/src/main/java/com/example/weathers_list_app/WeatherAdapter.java
package com.example.weathers_list_app;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;
import java.util.List;
import java.util.zip.Inflater;

public class WeatherAdapter extends RecyclerView.Adapter<WeatherAdapter.WeatherHolder> {

    List<Weather> weathers = new ArrayList<>();

    @NonNull
    @Override
    public WeatherHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.weather_item, parent,false);
        return new WeatherHolder(itemView);
    }

    @Override
    public void onBindViewHolder(@NonNull WeatherHolder holder, int position) {
        Weather w = weathers.get(position);
        holder.cityNameTextView.setText(w.getName());
        holder.temperatureTextView.setText(String.format("%.2f",w.getTemp()));
    }

    public void setWeathers(List<Weather> weathers){
        this.weathers = weathers;
        notifyDataSetChanged();
    }

    @Override
    public int getItemCount() {
        return weathers.size();
    }

    public Weather getWeatherAt(int position) {
        return weathers.get(position);
    }

    class WeatherHolder extends RecyclerView.ViewHolder{
        private TextView cityNameTextView;
        private TextView temperatureTextView;

        public WeatherHolder(@NonNull View itemView) {
            super(itemView);

            cityNameTextView = itemView.findViewById(R.id.cityNameTextView);
            temperatureTextView = itemView.findViewById(R.id.temperatureTextView);
        }
    }

}

File: NOT_MINE/WeatherAppList/app/src/main/java/com/example/weatherapplist/WeatherAdapter.kt
package com.example.weatherapplist

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView


class WeatherAdapter(private val list: List<WeatherList>)
    : RecyclerView.Adapter<WeatherViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): WeatherViewHolder {
        val inflater = LayoutInflater.from(parent.context)
        return WeatherViewHolder(inflater, parent)
    }

    override fun onBindViewHolder(holder: WeatherViewHolder, position: Int) {
        val weather: WeatherList = list[position]
        holder.bind(weather)
    }

    override fun getItemCount(): Int = list.size

}